### PR TITLE
Bump github.com/emicklei/go-restful version to v2.15.0

### PR DESCRIPTION
**Pull request by ash2k**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

I'm using a library that depends on k/k (yes, I know. Working on fixing that https://github.com/argoproj/gitops-engine/issues/56). When I build my project with Bazel I get the following warning:
```
DEBUG: /path/external/bazel_gazelle/internal/go_repository.bzl:189:18: com_github_emicklei_go_restful: gazelle: finding module path for import github.com/mjibson/appstats: exit status 1: go: finding module for package github.com/mjibson/appstats
cannot find module providing package github.com/mjibson/appstats: module github.com/mjibson/appstats: git ls-remote -q origin in /path/external/bazel_gazelle_go_repository_cache/pkg/mod/cache/vcs/444aab7ea271b6130ec26c66b91d9ad882ee59ce867f2107d31a0875343c64c2: exit status 128:
        fatal: could not read Username for 'https://github.com': terminal prompts disabled
Confirm the import path was entered correctly.
If this is a private repository, see https://golang.org/doc/faq#git_https for additional information.
```

This was because `github.com/emicklei/go-restful` depended on `github.com/mjibson/appstats`, that has been deleted. The issue has been fixed (https://github.com/emicklei/go-restful/issues/448 via https://github.com/emicklei/go-restful/commit/0d68c6d9ee3ced59b627f88f5e91bcef77a60481). Would be great to update to the version with the fix so that the warning goes away.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

Docs for dependency bump process: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/vendor.md

### CODE REVIEW

**Comment by ash2k** [comment]
/sig api-machinery

I'm not sure what sig to use here, please update it if my guess is incorect.

**Comment by lavalamp** [comment]
I don't see a reference to "github.com/mjibson/appstats" in the existing code that is being removed, which I guess is why it is working for us. I'm not sure why our own tools don't seem to care, maybe @liggitt would know.

I'm a little nervous about taking all the changes without looking more carefully at them.

**Comment by ash2k** [comment]
> I'm not sure why our own tools don't seem to care

I've noticed that too. My understanding is that `go mod vendor` vendors only the packages that are actually used in the codebase. The `examples` subpackage of `github.com/emicklei/go-restful` had that invalid import but Kubernetes does not use that (`examples`), so it's not vendored. In Kubernetes Bazel uses the vendored packages and does not download them. In my project what to download (`go_repository` bazel repo targets) are generated by Gazelle, which I guess is not smart enough to understand that the `examples` package (and hence everything that it imports) is not required, so I get the `go_repository` target with the `github.com/mjibson/appstats` dependency, which has been deleted.

File: vendor/github.com/emicklei/go-restful/container.go [hunk 1]
```diff
@@ -185,6 +185,11 @@ func logStackOnRecover(panicReason interface{}, httpWriter http.ResponseWriter)
 // when a ServiceError is returned during route selection. Default implementation
```
Comment (liggitt) [inline vendor/github.com/emicklei/go-restful/container.go:185]: this file is the only one that had changes that looked like they might impact us... I'd suggest a close review of the compression and handler type-checking

**Review by liggitt — COMMENTED** [review]
> I'm not sure why our own tools don't seem to care, maybe @liggitt would know.

a lot of the go mod tooling is sparse, and we use goproxy which insulates us some from unavailable modules/versions which used to be available

**Comment by ash2k** [comment]
So what are the next steps here? Nothing happened in almost a month =)

**Comment by lavalamp** [comment]
> So what are the next steps here? Nothing happened in almost a month =)

This is in my queue but it's extremely hard to reason about whether the changes made will affect the system and whether all such effects would be caught by tests. Sadly this has not been winning the competition for a large block of my time :/

**Comment by ash2k** [comment]
5 months and 4 rebases and still counting... I understand it's probably nobody's job to review and merge, but it's just demotivating. Why would I try to improve things if nothing from my recent PRs gets merged? I wonder if contributor experience sig is planning to do something to address this situation.

**Comment by BenTheElder** [comment]
>  I wonder if contributor experience sig is planning to do something to address this situation.

unfortunately that sig is one of the hardest hit out of the past year and running pretty light on contributors :/

there are just a lot of these and a very small group of maybe 3-5 active people reviewing dependency changes, it's hard to keep up : (

**Review by BenTheElder — APPROVED** [review]
https://github.com/kubernetes/kubernetes/pull/97961/files#r557004785 probably still needs deeper review, but licensing etc., no problematic `init`, etc. looks good.

cc @kubernetes/sig-architecture-pr-reviews to take a look at this one file in particular

**Comment by BenTheElder** [comment]
the robot will continue to state that this test failed because it no longer exists (the bazel job), it's a known design flaw :/

all tests are green now.

**Comment by BenTheElder** [comment]
cc @kubernetes/sig-architecture-pr-reviews

**Comment by BenTheElder** [comment]
bah wrong auto-completition, sorry! :/
cc @kubernetes/sig-api-machinery-pr-reviews

**Comment by lavalamp** [comment]
@ash2k please feel free to give your reviewer ([me in this case](https://github.com/kubernetes/kubernetes/pull/97961#issuecomment-775325602)) a friendly ping if you feel that more than a reasonable amount of time has passed. Which clearly it has in this case. Clearly I need to ask someone else to do this review since it isn't sorting to the top of my list, the problem is I'm not sure who can do it. I think the reason it's not sorting to the top of my list is the ratio of difficulty to lack of a pressing need.

**Comment by ash2k** [comment]
Ok, after 8 months of waiting, I think this is not going to happen. No point in rebasing.

**Pull request closed without merging**